app/assets/javascripts/admin/enterprises/controllers/enterprise_controller.js.coffee

```diff
@@ -90,3 +92,5 @@ angular.module("admin.enterprises")
 
     $scope.translation = (key) ->
       t('js.admin.enterprises.form.images.' + key)
+
+    RequestMonitor.load $scope.suppliers = Enterprises.index(action: "visible", ams_prefix: "basic", "q[is_primary_producer_eq]": "true")
```

Comment:
I dont think this is the correct way to fetch enterprises and I dont think this is the correct angular controler to load them.

I think the best pace to tackle this problem is here
app/assets/javascripts/darkswarm/controllers/products_controller.js.coffee
where the products are fetched and the filters applied.

In terms of how to load the producers, you already have $scope.order_cycle that maybe already has some form of $scope.order_cycle.suppliers list , otherwise you could build it here in a similar way to how $scope.supplied_taxons is built.
another option could be to simply use  $scope.Products and fetch the suppliers of each product.

But wait! Why do you need the producers here??? The ordering should all be done in the backend, you are already doing it in taxon_order in the renderer... can you explain what needs to be done here?

Reply:
The reasoning for this was to make the list of producers available to `producer_autocomplete`, so a list shows up when the user clicks the selector on the shop preferences page.

Initially I just built a list of all producers, but that should be filtered to get only relevant producers (it was suggested to get all producers included in the current order cycle, but I think there can be multiple open order cycles, so maybe we should get all producers included in any open order cycle).

But I do feel like this in not an appropriate way of doing that, I'm just trying to figure out how it should be done instead. If you have some tips please let me know :)

Reply:
oh, we are in the backoffice here :see_no_evil:  sorry, I missed that.

It would be incorrect to use OCs here imo, I think you actually doing it right, are you using 
the visible option here: app/controllers/admin/enterprises_controller.rb:168 ?

that should be the correct list of enterprises to show here filtered by is_producer :+1: 

And I think this is actually the right place to do this. So, is there something not working?

Reply:
Hey Luis, thanks, I was actually testing incorrectly, it seems like the the producer are being filetered correctly according to permissions! I just have a question regarding this (/app/assets/javascripts/admin/enterprises/controllers/enterprise_controller.js.coffee:96):
```
RequestMonitor.load $scope.suppliers = Enterprises.index(action: "visible", ams_prefix: "basic", "q[is_primary_producer_eq]": "true")
```
is the "visible" option you mentioned above what is being refered to by the `action: "visible"` param? Removing this param didn't seem to change the behavor (I still got the filtered producers), so what does it do? Also it seems like `ams_prefix` defaults to `"basic"`, so it could be removed too. Should I remove both parameters?
```
RequestMonitor.load $scope.suppliers = Enterprises.index("q[is_primary_producer_eq]": "true")
```